app/main/posts/modify/post-data-editor.directive.js

```diff
@@ -392,11 +392,13 @@ function PostDataEditorController(
 
             // Avoid messing with original object
             // Clean up post values object
+
             if ('message_location' in $scope.post.values) {
                 $scope.post.values.message_location = [];
             }
-            var post = PostEditService.cleanPostValues(angular.copy($scope.post));
 
+
+            var post = PostEditService.cleanPostValues(_.clone($scope.post));
```

Comment:
Why _.clone? Is it better?

Reply:
_.clone does not mess everything up. angular.copy basically removes all values from post.values :(  . I wish I had a better explanation.